src/kixi/datastore/dynamodb.clj

```diff
@@ -216,18 +235,49 @@
        (map (comp inflate-map (partial map-keys name)))
        (remove ::md/tombstone)))
 
+(defmulti select-key (fn [_ k] (type k)))
+(defmethod select-key nil [_ _])
+(defmethod select-key
+  clojure.lang.Keyword
+  [m k]
+  (select-keys m [k]))
+
+(defmethod select-key
+  clojure.lang.PersistentVector
+  [m ks]
+  {:pre [(every? keyword? ks)]}
+  (let [r (select-key m (first ks))]
+    (if (next ks)
+      (update r (first ks) select-key (vec (next ks)))
+      r)))
+
+(defn local-projection
+  [ks m]
+  (reduce (fn [a k]
+            (merge a (select-key m k))) {} ks))
+
```

Comment:
This is because we can not both execute a filter and select the values we return.  In the case in which we have a filter we return all the fields and then do a local projection.  This obviously gets very expensive the more columns and rows a table has.